common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/SierraService.scala

```diff
@@ -3,174 +3,130 @@ package uk.ac.wellcome.platform.stacks.common.services
 import java.time.Instant
 
 import akka.actor.ActorSystem
-import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.Uri
 import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials}
-import akka.http.scaladsl.model.{HttpMethods, HttpRequest}
-import akka.http.scaladsl.unmarshalling.Unmarshal
 import akka.stream.ActorMaterializer
-import cats.instances.future._
-import cats.instances.list._
-import cats.syntax.traverse._
-import io.circe.Json
-import uk.ac.wellcome.platform.sierra
-import uk.ac.wellcome.platform.sierra.ApiException
-import uk.ac.wellcome.platform.sierra.api.{V5itemsApi, V5patronsApi}
-import uk.ac.wellcome.platform.sierra.models.{ErrorCode, Hold, PatronHoldPost}
 import uk.ac.wellcome.platform.stacks.common.models._
 
-import scala.collection.JavaConverters._
-import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success, Try}
+import scala.concurrent.Future
 
-class SierraService(baseUrl: Option[String], username: String, password: String)(
+class SierraService(val baseUri: Uri = Uri(
+  "https://libsys.wellcomelibrary.org/iii/sierra-api"
+), credentials: BasicHttpCredentials)(
   implicit
-  as: ActorSystem,
-  am: ActorMaterializer,
-  ec: ExecutionContext
-) {
-  val sierraApiClient = new sierra.ApiClient()
-
-  baseUrl.foreach {
-    sierraApiClient.setBasePath
-  }
+    val system: ActorSystem,
+    val mat: ActorMaterializer
+) extends AkkaClientService
+    with AkkaClientServiceGet
+    with AkkaClientServicePost
+    with AkkaClientTokenExchange {
 
-  private val tokenUrlBase =
-    baseUrl.getOrElse(sierraApiClient.getBasePath)
+  import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+  import io.circe.generic.auto._
 
-  object AuthToken {
+  override val tokenPath = "v5/token"
 
-    import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+  def getItemStatus(sierraId: SierraItemIdentifier): Future[StacksItemStatus] =
+    for {
+      token <- getToken(credentials)
+      item <- get[SierraItemStub](
+        path = s"v5/items/$sierraId",
+        headers = List(Authorization(token))
+      )
 
-    val tokenPath = "v5/token"
-    val authHeader = Authorization(BasicHttpCredentials(username, password))
+    } yield StacksItemStatus(item.status.code)
 
-    val tokenRequest = HttpRequest(
-      method = HttpMethods.POST,
-      uri = f"$tokenUrlBase/$tokenPath",
-      headers = List(authHeader),
+  def placeHold(
+    userIdentifier: StacksUser,
```

Comment:
nit: the class name and the parameter don't match. Looking around, should `StacksUser` be `StacksUserIdentifier`?